src/partition/repart.c

```diff
@@ -0,0 +1,2983 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <fcntl.h>
+#include <getopt.h>
+#include <libfdisk.h>
+#include <linux/fs.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <sys/file.h>
+
+#include <openssl/hmac.h>
+#include <openssl/sha.h>
+
+#include "sd-id128.h"
+
+#include "alloc-util.h"
+#include "blkid-util.h"
+#include "blockdev-util.h"
+#include "btrfs-util.h"
+#include "conf-files.h"
+#include "conf-parser.h"
+#include "def.h"
+#include "efivars.h"
+#include "errno-util.h"
+#include "fd-util.h"
+#include "format-table.h"
+#include "format-util.h"
+#include "fs-util.h"
+#include "gpt.h"
+#include "id128-util.h"
+#include "list.h"
+#include "locale-util.h"
+#include "main-func.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "pretty-print.h"
+#include "proc-cmdline.h"
+#include "sort-util.h"
+#include "stat-util.h"
+#include "stdio-util.h"
+#include "string-util.h"
+#include "strv.h"
+#include "terminal-util.h"
+#include "utf8.h"
+
+/* Note: When growing and placing new partitions we always align to 4K sector size. It's how newer hard disks
+ * are designed, and if everything is aligned to that performance is best. And for older hard disks with 512B
+ * sector size devices were generally assumed to have an even number of sectors, hence at the worst we'll
+ * waste 3K per partition, which is probably fine. */
+
+static bool arg_dry_run = true;
```

Comment:
hmm? We dry-run by default. It's too easy to destroy your disk with partitioning tools so I figured you have to turn off dry-run before it actually does stuff. To make this discoverable the tool will suggest you to do this if you invoke it without --dry-run= specified.

Yes, it's the only tool that operates like that, but it's uniquely dangerous I would say...

Reply:
People might want to use this as alternative to sfdisk (because the syntax is easier to grok, and the weight model makes things nicely flexible). Having to specify dry-run always would be annoying. But let's leave it as is for now.

Reply:
The `dsctl` tool from 389ds has `--do-it` switch for removing the LDAP database. Maybe something like this? It is easier to type than `--dry-run=no` and it transports the meaning that no action will be taken by default without it